compiler/rustc_ty_utils/src/assoc.rs

```diff
@@ -345,13 +343,51 @@ fn impl_associated_item_for_impl_trait_in_trait(
         fn_has_self_parameter: false,
     });
 
+    impl_assoc_ty.param_env(tcx.param_env(impl_fn_def_id));
```

Comment:
Comment pls

Reply:
Something like "Our RPITIT inherits the param_env of the fn <why>"

Possible justifications are "because it has no extra predicates to assume" or something like that, though make it make sense :smile: